tests/assembleCoaddTestUtils.py

```diff
@@ -0,0 +1,625 @@
+# This file is part of pipe_tasks.
+#
+# LSST Data Management System
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+# See COPYRIGHT file at the top of the source tree.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+"""Set up simulated test data and simplified APIs for AssembleCoaddTask
+and its derived classes.
+
+This is not intended to test accessing data with the Butler and instead uses
+mock Butler data references to pass in the simulated data.
+"""
+from astropy.time import Time
+from astropy import units as u
+from astropy.coordinates import SkyCoord, EarthLocation, Angle
+import numpy as np
+
+from lsst.afw.cameraGeom.testUtils import DetectorWrapper
+import lsst.afw.geom as afwGeom
+import lsst.afw.image as afwImage
+import lsst.afw.image.utils as afwImageUtils
+from lsst.daf.butler import DeferredDatasetHandle
+from lsst.daf.persistence.dataId import DataId
+import lsst.geom as geom
+from lsst.geom import arcseconds, degrees
+from lsst.meas.algorithms.testUtils import plantSources
+from lsst.obs.base import MakeRawVisitInfoViaObsInfo
+import lsst.pipe.base as pipeBase
+from lsst.pipe.tasks.coaddInputRecorder import CoaddInputRecorderTask, CoaddInputRecorderConfig
+
+from astro_metadata_translator import makeObservationInfo
+
+__all__ = ["MockExposureReference", "makeMockSkyInfo", "MockCoaddTestData",
+           "MockGen2ExposureReference"]
+
+
+class MockGen2ExposureReference:
+    """Very simple object that looks like a Gen2 data reference to an exposure.
+
+    Attributes
+    ----------
```

Comment:
I wanted to document the class attributes, since I personally find that very helpful. The Attributes section is part of numpydoc even if it's not in our python doc guide, but I can change this to a `Parameters` section and move the remaining docstrings to in-line comments where the attributes are defined.